pkg/api/groups.go

```diff
@@ -59,10 +59,21 @@ var (
 	// itself. An alternative is to use atomic loads instead of RLock()
 	// and using atomic stores inside Lock() of a normal Mutex to serialize
 	// the writes (or use channel handshakes instead of a mutex).
-	cachedGroups     map[GroupDescriptor]string
-	cachedGroupsLock sync.RWMutex
+	cachedGroups                    map[GroupDescriptor]string
+	cachedGroupsLock                sync.RWMutex
+	cachedGroupVersionCount         map[groupDurationCacheKey]groupVersionCountCache
```

Comment:
You could initialize the map here:
```suggestion
	cachedGroupVersionCount         = make(map[groupDurationCacheKey]groupVersionCountCache)
```

That way, you could drop the `nil` check in `GetGroupVersionCountTimeline`.